File: csharp/all-your-base/AllYourBase.cs
using System;
using System.Collections.Generic;

public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        if (inputBase <= 1)
        {
            throw new ArgumentException("Input base cannot be less than 2");
        }

        if (outputBase <= 1)
        {
            throw new ArgumentException("Output base cannot be less than 2");
        }
        int sum = 0;
        for (int i = 0; i < inputDigits.Length; i++)
        {
            int digit = inputDigits[inputDigits.Length - 1 - i];
            if (digit < 0)
            {
                throw new ArgumentException("Digits cannot be less than 0");
            }

            if (digit >= inputBase)
            {
                throw new ArgumentException("Digits cannot be more than or equal with the input base");
            }
            
            sum += ((int)Math.Pow(inputBase, i)) * digit;
        }

        int max = 0;
        while (sum / (int)Math.Pow(outputBase, max + 1) >= 1)
        {
            max++;
        }

        int[] output = new int[max + 1];

        for (int i = 0; i < output.Length; i++)
        {
            int baseVal = (int)Math.Pow(outputBase, output.Length - 1 - i);
            output[i] = sum / baseVal;
            sum -= output[i] * baseVal;
        }

        return output;
    }
}

File: cpp/all-your-base/all_your_base.cpp
#include "all_your_base.h"
#include <stdexcept>
#include <algorithm>

namespace all_your_base
{

    std::vector<unsigned int> convert(unsigned int base, const std::vector<unsigned int> &digits, unsigned int output_base)
    {
        if (base < 2 || output_base < 2)
        {
            throw std::invalid_argument("Invalid base");
        }

        if (digits.empty())
        {
            return {};
        }

        unsigned long long value = 0;

        for (unsigned int digit : digits)
        {
            if (digit >= base)
                throw std::invalid_argument("Invalid digits");

            value = value * base + digit;
        }

        if (value == 0)
            return {};

        std::vector<unsigned int> result;

        while (value > 0)
        {
            result.push_back(value % output_base);
            value /= output_base;
        }

        std::reverse(result.begin(), result.end());

        return result;
    }

} // namespace all_your_base
